examples/images_contours_and_fields/contours_in_optimization_demo.py

```diff
@@ -0,0 +1,63 @@
+"""
+=============================
+Contours In Optimization Demo
+=============================
+
+Contour plotting is particularly handy when illustrating the solution
+space of optimization problems.  Not only can `.axes.Axes.contour` be
+used to represent the topography of the objective function, it can be
+used to generate boundary curves of the constraint functions.  The
+constraint lines can be drawn with
+`~matplotlib.patheffects.TickedStroke` to distinguish the valid and
+invalid sides of the constraint boundaries.
+
+`.axes.Axes.contour` generates curves with larger values to the left
+of the contour.  Consequently, when using
+`~matplotlib.patheffects.TickedStroke` to illustrate a constraint in
+a typical optimization problem, the angle should be set between
+zero and 180 degrees.
```

Comment:
To point to the lower or higher value? I think this is implied, or something you might know as someone who works on optimization problems, but not explicitly stated.

Reply:
Trying to strike the right balance here...  An earlier revision was deemed to have too much domain jargon.  A big part of the solution to that was to spool off this separate file under the context of optimization.  So, perhaps dialing up the domain specific jargon is appropriate here.

Just like putting a quadratic equation in the form of ax^2+bx+c=0 (with the sign rolled into the coefficients making a,b,c appear positive), optimization problems have a standard form.  Most numeric optimizers use the same standard form -- that way it is easy to implement your problem in someone else's optimizer.  Just like implementing a quadratic solver as [ans1,ans2]=quadeq(a,b,c) or something.

For inequality constraints, the constraint functions are usually called g(x).  The standard form of the inequality constraint is g(x)<=0.  This is called the negative-null form because negative and zero values of g(x) satisfy the constraint.  So, you draw the constraint line at g(x)=0 and draw the ticks to the bad side -- to the side of g(x)>0 i.e. increasing g(x).  So, the ticks 'point' at the higher value.

The matplotlib constraint code consistently generates contours such that the function is increasing on the 'left' side of the line -- which with tickedstroke means angles from 0 to 180.

Reply:
I don't think we need so much detail.

The previous sentence says 'curves with larger values to the left of the contour'. This sentence says angle should be between 0 and 180. You are missing a link between the two. Either state that you're pointing at larger values, or state which direction the angle is measured.

In other words, this example description has to stand alone. Yes, the direction of the angle is in the docstring, but don't expect anyone to look at the docstring if they can't figure out what the example is about.

Reply:
Should be fixed in 275d0b2e.  Let me know if that clarifies things sufficiently.